CRM/Contact/Form/GroupContact.php

```diff
@@ -73,7 +73,7 @@ public function buildQuickForm() {
         if ($onlyPublicGroups && $group['visibility'] == 'User and User Admin Only') {
           continue;
         }
-        $allGroups[$id] = $group;
+        $allGroups[$group['id']] = $group;
```

Comment:
Could you discard the `as $id` from the foreach loop above? It will avoid someone accidentally changing it back to what it was before.